packages/graphql-codegen-cli/src/utils/watcher.ts

```diff
@@ -99,22 +98,20 @@ export const createWatcher = (
       async (_, events) => {
         // it doesn't matter what has changed, need to run whole process anyway
         await Promise.all(
+          // NOTE: @parcel/watcher always provides path as an absolute path
           events.map(async ({ type: eventName, path }) => {
-            /**
-             * @parcel/watcher has no way to run watcher on specific files (https://github.com/parcel-bundler/watcher/issues/42)
-             * But we can use micromatch to filter out events that we don't care about
-             */
-            if (!mm.contains(path, files)) return;
+            if (!shouldRebuild({ path })) {
+              return;
+            }
 
             lifecycleHooks(config.hooks).onWatchTriggered(eventName, path);
             debugLog(`[Watcher] triggered due to a file ${eventName} event: ${path}`);
-            const fullPath = join(watchDirectory, path);
             // In ESM require is not defined
             try {
-              delete require.cache[fullPath];
+              delete require.cache[path];
             } catch (err) {}
 
-            if (eventName === 'update' && config.configFilePath && fullPath === config.configFilePath) {
+            if (eventName === 'update' && config.configFilePath && path === config.configFilePath) {
               log(`${logSymbols.info} Config file has changed, reloading...`);
```

Comment:
There was no need for `fullPath` here, because `path` is already absolute, and so concatenating it with `watchDirectory` was resulting in invalid paths (combining two absolute paths together). We can just use `path` instead. See f6ab51112f5fec9b6e8a7a0e5bed974bdc525420